src/Uno.Extensions.Navigation.UI/Navigators/FrameNavigator.cs

```diff
@@ -109,101 +96,167 @@ protected override bool CanNavigateToRoute(Route route)
-				if (entry is null ||
-					entry.SourcePageType != seg.Map.RenderView)
-				{
-					var newEntry = new PageStackEntry(seg.Map.RenderView, null, null);
-					Control?.BackStack.Add(newEntry);
-				}
-				firstSegment = firstSegment.Append(segments[i + 1].Route);
-				route = route.Trim(seg.Route);
-			}
 
-			// Trim any excess backstack values
-			while (Control?.BackStack.Count > segments.Length - 1)
-			{
-				Control.BackStack.RemoveAt(Control.BackStack.Count - 1);
-			}
+		// Need to navigate the underlying frame if it's not already
+		// displaying the correct page
+		if (Control!.SourcePageType != lastMap?.RenderView)
+		{
+			await Show(lastRoute.Base, lastMap?.RenderView, lastRoute.Data);
 		}
 		else
 		{
-			var numberOfPagesToRemove = route.FrameNumberOfPagesToRemove();
-			// We remove 1 less here because we need to remove the current context, after the navigation is completed
-			while (numberOfPagesToRemove > 1)
+			// Rebuild the nested region hierarchy
+			Control.ReassignRegionParent();
+			if (segments.Length > 1 ||
+				string.IsNullOrWhiteSpace(request.Route.Path))
 			{
-				RemoveLastFromBackStack();
-				numberOfPagesToRemove--;
+				refreshViewModel = true;
 			}
+		}
 
-			for (var i = 0; i < segments.Length - 1; i++)
-			{
-				var seg = segments[i];
-				var newEntry = new PageStackEntry(seg.Map.RenderView, null, null);
-				Control?.BackStack.Add(newEntry);
-				route = route.Trim(seg.Route);
-				firstSegment = firstSegment.Append(segments[i + 1].Route);
-			}
 
-			await Show(segments.Last().Route.Base, segments.Last().Map.RenderView, route.Data);
+		// If path starts with / then remove all prior pages and corresponding contexts
+		if (route.FrameIsRooted())
+		{
+			ClearBackStack();
+		}
+
+		// If there were pages to remove, after navigating we need to remove
+		// the page that we've navigated away from.
+		if (route.FrameNumberOfPagesToRemove() > 0)
+		{
+			RemoveLastFromBackStack();
+		}
 
-			// If path starts with / then remove all prior pages and corresponding contexts
-			if (route.FrameIsRooted())
-			{
-				ClearBackStack();
-			}
 
-			// If there were pages to remove, after navigating we need to remove
-			// the page that we've navigated away from.
-			if (route.FrameNumberOfPagesToRemove() > 0)
+		Route? firstSegment = null;
+		for (var i = 0; i < segments.Length - 1; i++)
+		{
+			var (r, map, isDependsOn) = segments[i];
+			if(r.IsEmpty() || (isDependsOn && (Control?.BackStack.Any(entry=>entry.SourcePageType==map?.RenderView)??false)))
 			{
-				RemoveLastFromBackStack();
+				continue;
 			}
-		}
 
-		await InitializeCurrentView(request, route, Resolver.Find(route), refreshViewModel);
+			var newEntry = new PageStackEntry(map?.RenderView, null, null);
+			Control?.BackStack.Add(newEntry);
+			firstSegment = firstSegment?.Append(r) ?? r;
+		}
+		firstSegment = firstSegment?.Append(lastRoute) ?? lastRoute;
+
+
+
+
+		//var firstSegment = segments.First().Route;
```

Comment:
Keeping commented code for prosperity?

Reply:
yep - will push a new PR tomorrow with clean up of commented code and adding informational comments where appropriate